de.fhdo.lemma.technology.technologydsl.extractor/src/de/fhdo/lemma/technology/technologydsl/extractor/TechnologyDslExtractor.xtend

```diff
@@ -0,0 +1,95 @@
+package de.fhdo.lemma.technology.technologydsl.extractor
+
+import de.fhdo.lemma.technology.DataFormat
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.technology.Technology
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.JoinPointType
+import de.fhdo.lemma.technology.TechnologySpecificPrimitiveType
+import de.fhdo.lemma.technology.ServiceAspect
+
+/**
+ * Model-to-text extractor for the Technology DSL.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class TechnologyDslExtractor {
+
+    /**
+     * Extract TechnologyModel
+     */
+    def String extractToString(Technology techModel) {
+        val types = if (techModel.primitiveTypes !== null) {
+            // Currently only supports primitiveTypes for generation
+            '''types {
+                «FOR pt : techModel.primitiveTypes SEPARATOR '\n'»«pt.extractToString»«ENDFOR»
```

Comment:
Check line length. See https://seelabfhdo.github.io/lemma-docs/developer-guide/coding-style/#indentation for more information.